lib/app.rb

```diff
@@ -11,59 +43,76 @@ def initialize
 
   def build_board
     loop do
-      board_rows_input = @display.get_board_rows
-      baord_columns_input = @display.get_board_columns
-      @simulator.setup_board(board_rows_input, baord_columns_input)
+      @display.print(MESSAGES[:enter_board_cols])
+      board_columns_input = @display.get_input_to_int
+
+      @display.print(MESSAGES[:enter_board_rows])
+      board_rows_input = @display.get_input_to_int
 
-      @display.build_board_complete_message
-      @display.print_board(@simulator.board.board_with_axis)
+      @simulator.setup_board(board_rows_input, board_columns_input)
+
+      @display.print(MESSAGES[:build_board_complete])
+      @display.draw(@simulator.board.board_with_axis)
 
       return if @simulator.board
     rescue ArgumentError => e
-      @display.print_error(e)
+      @display.print(e)
       next
     end
   end
 
   def place_robot
-    @display.start_place_robot_message
+    @display.print(MESSAGES[:start_place_robot])
 
     loop do
-      robot_column_input = @display.get_robot_start_column
-      robot_row_input = @display.get_robot_start_row
-      robot_bearing_input = @display.get_robot_start_bearing
+      @display.print(MESSAGES[:enter_robot_start_col])
+      robot_column_input = @display.get_input_to_int
+
+      @display.print(MESSAGES[:enter_robot_start_row])
+      robot_row_input = @display.get_input_to_int
+
+      @display.print(MESSAGES[:enter_robot_bearing])
+      robot_bearing_input = @display.get_input_to_sym
+
       @simulator.place(x: robot_column_input, y: robot_row_input, direction: robot_bearing_input)
 
-      @display.show_robot_starting_position_message
-      @display.print_board(@simulator.board.board_with_axis)
+      @display.print(MESSAGES[:robot_starting_position_message])
+      @display.draw(@simulator.board.board_with_axis)
 
       return if @simulator.robot.bearing
     rescue ArgumentError => e
-      @display.print_error(e)
+      @display.print(e)
       next
     end
   end
 
   def run_commands
-    @display.show_run_command_intro_message
+    @display.print(MESSAGES[:run_command_intro_message])
 
     loop do
-      @display.show_command_instructions
+      @display.print(MESSAGES[:command_instructions])
+      @display.print(MESSAGES[:enter_command])
 
-      commands = @display.get_user_commands
+      commands = @display.get_input_as_str
       system 'clear'
-      @simulator.evaluate(commands) { |outcome| @display.print_outcome(@simulator.robot, @simulator.board, outcome) }
-      @display.print_board(@simulator.board.board_with_axis)
-      @display.enter_another_command
+
+      @simulator.evaluate(commands) do |command|
+        @display.print("#{MESSAGES[:current_command]} #{command}\n")
+        @display.draw(@simulator.board.board_with_axis)
+        @display.print("#{MESSAGES[:robot_coordinates]} #{@simulator.robot.coordinates}")
+        @display.print("#{MESSAGES[:robot_bearing]} #{@simulator.robot.bearing}")
```

Comment:
Based on your [comment](https://github.com/j-tws/toy-robot-ruby/pull/1#discussion_r1404711314), this will be a series of notifications that uses the `display` generic functions which prints out relevant information as each command is being evaluated.